src/TuxStoreMixinGenerator.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+
+//TuxStoreMixinGenerator FUNCTION: creates an object with the componentDidMount and componentWillUnmount methods that will add and remove the specified event listeners from the provided store
+//@param props OBJECT: TuxStore for which actions will be registered, needed for waitFor syntax [ALTERNATE ARRAY: array of objects with same keys as listed below]
+  //expected keys:
+  // store OBJECT: store object that the event and listener should be attached to
+  // listener FUNCTION: callback function to be invoked upon associated event [ALTERNATE ARRAY: array of callback functions]
+  // event STRING: event type to trigger listener callback function [ALTERNATE ARRAY: array of event type strings]
+var TuxStoreMixinGenerator = function (props) {
+  var storeConnections = {};
+  // *****Should be able to to refactor this to check if props is not an
+  // array and to then turn it into an array such that the code is dried up
+  // check to see if receiving an array with multiple objects
+  if (Array.isArray(props)) {
+    var propsLength = props.length;
+
+    // add the events and listeners for the particular store to the componentDidMount React life cycle event
+    storeConnections.componentDidMount = function () {
+      for (var i = 0; i < propsLength; i++){
+        // map addChangeListeners, due to the second input input argument set to true, to store, listeners, and events passed in
+        mapListenersAndEventsToStore(props[i], true);
+      }
+    };
+
+    // add the events and listeners for the particular store to the componentDidMount React life cycle event
+    storeConnections.componentWillUnmount = function() {
+      for (var i = 0; i < propsLength; i++){
+        // map addChangeListeners, due to the second input argument set to false, to store, listeners, and events passed in
+        mapListenersAndEventsToStore(props[i], false);
+      }
+    };
+  } else if (typeof props === 'object' && props !== null ) {
+    // add the events and listeners for the particular store to the componentDidMount React life cycle event
+    storeConnections.componentDidMount = function () {
+      // map addChangeListeners, due to the second input input argument set to true, to store, listeners, and events passed in
+      mapListenersAndEventsToStore(props, true);
+    };
+
+    // add the events and listeners for the particular store to the componentDidMount React life cycle event
+    storeConnections.componentWillUnmount = function() {
+      // map addChangeListeners, due to the second input argument set to false, to store, listeners, and events passed in
+      mapListenersAndEventsToStore(props, false);
+    };
+  }
+
+  return storeConnections;
+};
+
+//mapListenersAndEventsToStore FUNCTION maps adding or removing change listeners from a store
+//@param storeConfig OBJECT: TuxStore for which actions will be registered, needed for waitFor syntax [ALTERNATE ARRAY: array of objects with same keys as listed below]
```

Comment:
i dont think this is needed for waitFor syntax, is it?  Also, this doesn't accept an array only an object right